core/drivers/crypto/hisilicon/hpre_ecc.c

```diff
@@ -0,0 +1,967 @@
+	0xA5, 0xD0, 0x3B, 0xB5, 0xC9, 0xB8, 0x89, 0x9C, 0x47, 0xAE, 0xBB, 0x6F,
+	0xB7, 0x1E, 0x91, 0x38, 0x64, 0x09
+};
+
+/* NID_SM2 */
+static uint8_t g_sm2_p[] = {
+	0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff
+};
+
+static uint8_t g_sm2_a[] = {
+	0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfc
+};
+
+static uint8_t g_sm2_b[] = {
+	0x28, 0xe9, 0xfa, 0x9e, 0x9d, 0x9f, 0x5e, 0x34, 0x4d, 0x5a, 0x9e, 0x4b,
+	0xcf, 0x65, 0x09, 0xa7, 0xf3, 0x97, 0x89, 0xf5, 0x15, 0xab, 0x8f, 0x92,
+	0xdd, 0xbc, 0xbd, 0x41, 0x4d, 0x94, 0x0e, 0x93
+};
+
+static uint8_t g_sm2_gx[] = {
+	0x32, 0xc4, 0xae, 0x2c, 0x1f, 0x19, 0x81, 0x19, 0x5f, 0x99, 0x04, 0x46,
+	0x6a, 0x39, 0xc9, 0x94, 0x8f, 0xe3, 0x0b, 0xbf, 0xf2, 0x66, 0x0b, 0xe1,
+	0x71, 0x5a, 0x45, 0x89, 0x33, 0x4c, 0x74, 0xc7
+};
+
+static uint8_t g_sm2_gy[] = {
+	0xbc, 0x37, 0x36, 0xa2, 0xf4, 0xf6, 0x77, 0x9c, 0x59, 0xbd, 0xce, 0xe3,
+	0x6b, 0x69, 0x21, 0x53, 0xd0, 0xa9, 0x87, 0x7c, 0xc6, 0x2a, 0x47, 0x40,
+	0x02, 0xdf, 0x32, 0xe5, 0x21, 0x39, 0xf0, 0xa0
+};
+
+static uint8_t g_sm2_n[] = {
+	0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0x72, 0x03, 0xdf, 0x6b, 0x21, 0xc6, 0x05, 0x2b,
+	0x53, 0xbb, 0xf4, 0x09, 0x39, 0xd5, 0x41, 0x23
+};
+
+static struct hpre_ecc_curve g_curve_list[] = {
+	{
+		.id = TEE_ECC_CURVE_NIST_P192,
+		.key_bits = 192,
+		.p = g_prime192v1_p,
+		.a = g_prime192v1_a,
+		.b = g_prime192v1_b,
+		.x = g_prime192v1_gx,
+		.y = g_prime192v1_gy,
+		.n = g_prime192v1_n,
+	}, {
+		.id = TEE_ECC_CURVE_NIST_P224,
+		.key_bits = 224,
+		.p = g_secp224r1_p,
+		.a = g_secp224r1_a,
+		.b = g_secp224r1_b,
+		.x = g_secp224r1_gx,
+		.y = g_secp224r1_gy,
+		.n = g_secp224r1_n,
+	}, {
+		.id = TEE_ECC_CURVE_NIST_P256,
+		.key_bits = 256,
+		.p = g_prime256v1_p,
+		.a = g_prime256v1_a,
+		.b = g_prime256v1_b,
+		.x = g_prime256v1_gx,
+		.y = g_prime256v1_gy,
+		.n = g_prime256v1_n,
+	}, {
+		.id = TEE_ECC_CURVE_NIST_P384,
+		.key_bits = 384,
+		.p = g_secp384r1_p,
+		.a = g_secp384r1_a,
+		.b = g_secp384r1_b,
+		.x = g_secp384r1_gx,
+		.y = g_secp384r1_gy,
+		.n = g_secp384r1_n,
+	}, {
+		.id = TEE_ECC_CURVE_NIST_P521,
+		.key_bits = 521,
+		.p = g_secp521r1_p,
+		.a = g_secp521r1_a,
+		.b = g_secp521r1_b,
+		.x = g_secp521r1_gx,
+		.y = g_secp521r1_gy,
+		.n = g_secp521r1_n,
+	}, {
+		.id = TEE_ECC_CURVE_SM2,
+		.key_bits = 256,
+		.p = g_sm2_p,
+		.a = g_sm2_a,
+		.b = g_sm2_b,
+		.x = g_sm2_gx,
+		.y = g_sm2_gy,
+		.n = g_sm2_n,
+	}
+};
+
+static bool is_all_zero(uint8_t *data, uint32_t size, const char *p_name)
+{
+	uint32_t i = 0;
+
+	for (i = 0; i < size; i++) {
+		if (data[i] > 0)
+			return false;
+	}
+
+	EMSG("Error: %s all zero", p_name);
```

Comment:
It looks like all these `EMSG()` should be `DMSG()` instead.

Reply:
I think this is error msg, and we use EMSG() in other files such as dh.c